com.unity.render-pipelines.universal/Runtime/2D/LightUtility.cs

```diff
@@ -57,6 +67,199 @@ private struct SpriteLightMeshVertex
+                    max = path[i].N;
+                    newMin = true;
+                }
+            }
+
+            List<IntPoint> sorted = new List<IntPoint>();
+            sorted.AddRange(path.GetRange(minIndex, (path.Count - minIndex)));
+            sorted.AddRange(path.GetRange(0, minIndex));
+            return sorted;
+        }
+        
+        // Ensure that all points eliminated due to overlaps and intersections are accounted for Tessellation.
+        static List<IntPoint> FixPivots(List<IntPoint> pathInput, int pathLength)
+        {
+            var path = SortPivots(pathInput);
+            long pivotPoint = path[0].N;
+            
+            // Connect Points for Overlaps.
+            for (int i = 1; i < path.Count; ++i)
+            {
+                var prev = path[i - 1];
+                var curr = path[i];
+
+                if (prev.N > curr.N)
+                {
+                    var incr = TestPivot(path, i, pivotPoint);
+                    if (incr)
+                    {
+                        var test = curr;
+                        test.N = (pivotPoint + 1) < pathLength ? (pivotPoint + 1) : 0;
+                        test.D = 3;
+                        path[i] = test;
+                    }
+                }
+                pivotPoint = path[i].N;
+            }
+
+            return path;
+        }
+
+        public static Bounds GenerateShapeMesh(Mesh mesh, Vector3[] shapePath, float falloffDistance)
+        {
+
+            var ix = 0;
+            var vcount = 0;
+            var icount = 0;
+            const float kClipperScale = 10000.0f;
+                
+            // todo Revisit this while we do Batching. 
+            var meshInteriorColor = new Color(0.0f,0,0,1.0f);
+            var meshExteriorColor = new Color(0.0f,0,0,0.0f);
+            var vertices = new NativeArray<ParametricLightMeshVertex>(shapePath.Length * 256, Allocator.Temp);
+            var indices = new NativeArray<ushort>(shapePath.Length * 256, Allocator.Temp);
+
+            // Create shape geometry
+            var innerShapeVertexCount = shapePath.Length;
+            var inner = new ContourVertex[innerShapeVertexCount + 1];
+            for (var i = 0; i < innerShapeVertexCount; ++i)
+                inner[ix++] = new ContourVertex() { Position = new Vec3() { X = shapePath[i].x, Y = shapePath[i].y, Z = 0 } };
+            inner[ix++] = inner[0];
+
+            var tess = new Tess();
+            tess.AddContour(inner, ContourOrientation.CounterClockwise);
+            Tessellate(tess, ElementType.Polygons, indices, vertices, meshInteriorColor, ref vcount, ref icount);
+
+            // Create falloff geometry
+            List<IntPoint> path = new List<IntPoint>();
+            var pointNoise = new System.Random();
+            for (var i = 0; i < innerShapeVertexCount; ++i)
+            {
+                var pnx = 0;// pointNoise.Next(10);
+                var pny = 0;// pointNoise.Next(10);
```

Comment:
Can we remove these?

Reply:
They were used for testing. Yes, these have been removed in the latest.